### PR TITLE
Fix refs in macros

### PR DESCRIPTION
**Pull request by cmcarthur**
Fixes snowplow package. Adds a test for refs in macros.

### CODE REVIEW

**Comment by drewbanin** [comment]
@cmcarthur this fix lgtm, but there might be more to do here. The existing implementation of dbt contains the following variables in compilation context:

- `{{ target }}`
- `{{ var(...)  }}`
- `{{ this }}`

There's an argument to be made that these don't belong in the macro context, and should instead be passed in through a macro argument. At present, this fails with a big stacktrace ending with 

```jinja2.exceptions.UndefinedError: 'target' is undefined```

At the very least, we should catch this error and show a more helpful message

File: dbt/parser.py [hunk 1]
```diff
@@ -145,6 +147,9 @@ def parse_node(node, node_path, root_project_config, package_project_config,
 
     context = {}
 
+    context['ref'] = __ref(parsed_node)
+    context['config'] = __config(parsed_node, config)
```
Comment (drewbanin) [inline dbt/parser.py]: can we add stubs for `target` and `var()` here?

File: dbt/parser.py [hunk 2]
```diff
@@ -145,6 +147,11 @@ def parse_node(node, node_path, root_project_config, package_project_config,
...
+    context['var'] = lambda x: ''
+    context['target'] = property(lambda x: '', lambda x: x)
```
Comment (cmcarthur) [inline dbt/parser.py]: @drewbanin check this out lol

File: test/integration/016_macro_tests/macros/my_macros.sql [hunk 3]
```diff
@@ -6,3 +6,13 @@
...
 
 {% endmacro %}
+
+
+{% macro with_ref() %}
+
+   {{ref('dep_macro')}}
+--   {{this}}
+--   {{var('test')}}
+--   {{target.name}}
```
Comment (cmcarthur) [inline test/integration/016_macro_tests/macros/my_macros.sql]: added tests for these

File: dbt/parser.py [hunk 4]
```diff
@@ -149,6 +149,8 @@ def parse_node(node, node_path, root_project_config, package_project_config,
 
     context['ref'] = __ref(parsed_node)
     context['config'] = __config(parsed_node, config)
+    context['var'] = lambda x: ''
```
Comment (drewbanin) [inline dbt/parser.py]: can we do `context['var'] = lambda *args: ''`

`var()` can optionally take a second arg, the default value if none is provided

File: dbt/parser.py [hunk 5]
```diff
@@ -149,6 +149,8 @@ def parse_node(node, node_path, root_project_config, package_project_config,
...
+    context['target'] = property(lambda x: '', lambda x: x)
```
Comment (drewbanin) [inline dbt/parser.py]: awesome, didn't know about `property`!

**Pull request merged**